core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -1086,7 +1133,7 @@ private[spark] class TaskSetManager(
           addPendingTask(index)
           // Tell the DAGScheduler that this task was resubmitted so that it doesn't think our
           // stage finishes when a total of tasks.size tasks finish.
-          sched.dagScheduler.taskEnded(
+          emptyTaskInfoAccumulablesAndNotifyDagScheduler(tid,
             tasks(index), Resubmitted, null, Seq.empty, Array.empty, info)
```

Comment:
+CC @cloud-fan here, we will be firing with incorrect `accumulables`.

Reply:
TBH, I think this API is not well-designed and we don't have a clear definition anyway. @JoshRosen 's comment explains the details: if a task is running, listeners won't get the `Resubmitted` event, but the task failure event. The `Resubmitted` event is only issued if the task is completed and its map output is lost. In this case, I think the current behavior to send the task info of the previously completed task info is a bit weird. It's likely not by design but just a coincidence. Why would people care the previously completed task's metrics when Spark resubmits it? Even if they care, they can get the metrics in the previous task completion event.

To me, the memory issue is more important to most users, and a listener that will be broken by this change should be very rare. How about we put it in the migration guide and ask people to set the legacy flag to get the legacy behavior?

Reply:
Yes, what @JoshRosen [detailed for executorLost](https://github.com/apache/spark/pull/44321#pullrequestreview-1785137821) is what I am referring to - had missed that comment, thanks for the analysis Josh !

Fixing memory issue is definitely something we should aspire for, but IMO not at the cost of breaking interfaces and existing user code - in this case, `Resubmitted` has existed for 8+ years; and has been exposed to developer's.
How commonly this is triggered depends on stability of the cluster infra - which might be different for different users/deployments :-)

In this specific case, unfortunately it impacts not just end user code (which we dont know is using it: and I have seen fairly involved usage of these api's, including writing some of my own in the past) - but also our spark UI - for example `AppStatusListener.onTaskEnd` when `Resubmitted` is fired.

If we want to relook at the behavior of how `Resubmitted` is handled in Spark, we should do that decoupled from a performance discussion - it is possible that some of this has evolved in "interesting ways" since initially designed, and requires to be revisited.
For this PR, which is to improve memory utilization, it should not have unintended side effects IMO.

As I mentioned before, at a minimum, this should not become default.
Once we relook/redesign `Resubmitted`, we can flip it - if required.

Reply:
I checked the SQL UI (`SQLAppStatusListener`) and the task metrics from `Resubmitted` is ignored because the partition is already marked as finished. Anyway, I'm fine to turn it off by default, but I don't really think the metrics of a `Resubmitted` event is useful. We can discuss it in a different thread though.